recipes-core/initrdscripts/files/ni_provisioning.common

```diff
@@ -217,11 +231,23 @@ elif [ "$ARCH" = "x86_64" ]; then
     find_target_block_device()
     {
         local selected_block=""
+        local mmc_ok=false
+
+        # Check if the machine's DeviceCode is in the MMC device whitelist.
+        if [ -e "${MMC_DEVICE_ALLOW_PATH}" ]; then
+                device_code=$(fw_printenv -n DeviceCode 2>/dev/null)
+                [ -n "$device_code" ] && \
+                        grep -iq -E "^\s*(0x)?${device_code}\s*$" "${MMC_DEVICE_ALLOW_PATH}" && \
```

Comment:
If you remove the *$ at the end of the regex, you'd match any line beginning with a string of zero or more spaces and a hexadecimal device code and ignore anything after it.

Reply:
I'll give that a whirl.